lib/deps/upsert.js

```diff
@@ -7,10 +7,6 @@ var Promise = require('./promise');
 // the doc, or false if it doesn't need to do an update after all
 var upsert = module.exports = function upsert(db, docId, diffFun) {
   return new Promise(function (fulfill, reject) {
-    if (typeof docId !== 'string') {
-      return reject(new Error('doc id is required'));
-    }
-
```

Comment:
I have no idea what this has to do with this PR, but it makes sense to me to remove it because it's only used internally, so I'm sure we're not passing in any non-string _ids anywhere.